nxapi/nxlog.py

```diff
@@ -1,22 +1,52 @@
+from datetime import datetime
+import logging
+import re
+import socket
+import copy
+import itertools
+
 try:
     from urlparse import parse_qs
 except ImportError:  # python3
     from urllib.parse import parse_qs
 
 from . import rules
 
+def parse_date(nxlog):
+    """
+    :param str nxlog: A naxsi log ( https://github.com/nbs-system/naxsi/wiki/naxsilogs )
+    :return string: date string or empty string if we fail to find a date
+    """
+    date_regex = re.compile("""(((Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)""" \
```

Comment:
There is no need to compile this regexp inside a function: either move it to the file's scope, so it'll be compiled on import, or use it in a non-compiled form.